uetools/UePostproc/Postproc.py

```diff
@@ -198,7 +198,337 @@ def calc_forcebalance(self, cutlo=1e-300):
+        Index associated with the component names above: (1,2,3,4,5,6,7,8).
+
+        Right-hand side of total momentum eqn is up_net(ix,iy,ifld).
+        Amplitude order of up_net components are in ordcmpname
+
+        """
+        from numpy import zeros,abs,empty,int
+        from sys import modules
+        self.momentum ={"momentum":{},"force":{},"amplitude":{},"rank":{}}
+
+        # Load all variables necessary to the local namespace
+        for var in ['ev', 'misotope', 'natomic', 'mi', 'zi', 'rrv', 'gpex', 
+            'gpix', 'gtex', 'gtix', 'pri', 'netap', 'qe', 'fqp', 'sx',
+            'alfe', 'loglambda', 'betai', 'ex', 'vol', 'up', 'volmsor',
+            'pondomfpari_use','nx','ny','nusp','ixm1','ixp1','fmix','fmixy',
+            'gxf','volv','ni','uu','fmiy','cpiup','qe','frici','fricnrl',
+            'cfmsor','msor','msorxr','psor','psorxr','iysptrx','dx','rr',
+            'nuiz','iigsp','cfneut','nm','cfupcx','nurc','nucx','fnix',
+            'fniy',
+        ]:
+            setattr(modules[__name__], var, self.get(var))       
+
+        # Set up necessary arrays
+        # Components of momentum eqn
+        fmox = zeros((nx+2,ny+2,nusp))
+        fmoxconv = zeros((nx+2,ny+2,nusp))
+        fmoxvis = zeros((nx+2,ny+2,nusp))
+        fmoy = zeros((nx+2,ny+2,nusp))
+        pridx = zeros((nx+2,ny+2,nusp))
+        efieldf = zeros((nx+2,ny+2,nusp))
+        thermf = zeros((nx+2,ny+2,nusp))
+        momfric = zeros((nx+2,ny+2,nusp))
+        momiz = zeros((nx+2,ny+2,nusp))
+        momcxrec = zeros((nx+2,ny+2,nusp))
+        sxv = zeros((nx+2,ny+2))
+
+        # Components of upar eqn
+        fmoxv = zeros((nx+2,ny+2,nusp))
+        fmoyv = zeros((nx+2,ny+2,nusp))
+        pridxv = zeros((nx+2,ny+2,nusp))
+        efieldfv = zeros((nx+2,ny+2,nusp))
+        thermfv = zeros((nx+2,ny+2,nusp))
+        momfricv = zeros((nx+2,ny+2,nusp))
+        momizv = zeros((nx+2,ny+2,nusp))
+        momcxrecv = zeros((nx+2,ny+2,nusp))
+
+       
+
+        # Sum of components
+        up_net = zeros((nx+2,ny+2,nusp))
+        mom_net = zeros((nx+2,ny+2,nusp))
+
+        xparrb = zeros((nx+2,ny+2))    # parallel distance from right bdry
+
+        # Compute components of momentum equation; these are resmo/volv
+        for iis in range(nusp):  # loop over all momentum eqns
+            for iy in range(1,ny+1):
+                for ix in range(1,nx):
+                    ix1 = ixm1[ix,iy]
+                    ix2 = ixp1[ix,iy]
+                    vratio = sx[ix,iy]/(gxf[ix,iy]*volv[ix,iy])
+            
+                    fmox[ix,iy,iis] = ( -(fmix[ix2,iy,iis] - fmixy[ix2,iy,iis]) + \
+                                (fmix[ix, iy,iis] - fmixy[ix, iy,iis]) )/volv[ix,iy]
+                    sxv[ix,iy] = 0.5*(sx[ix1,iy]+sx[ix,iy])
+                    sxv[ix2,iy] = 0.5*(sx[ix,iy]+sx[ix2,iy])
+                    fmoxconv[ix,iy,iis] =mi[iis]*(-ni[ix2,iy,iis]*uu[ix2,iy,iis]*up[ix2,iy,iis]* \
```

Comment:
The fmoxconv-term should be the term corresponding to fmox, only calculated externally